From: Ignacio&nbsp;GalarzaDate: December 14 2006 10:41pm
Subject: bk commit into 5.0 tree (iggy:1.2355) BUG#18156

Below is the list of changes that have just been committed into a local
5.0 repository of iggy. When iggy does a push these changes will
be propagated to the main repository and, within 24 hours after the
push, to the public repository.
For information on how to access the public repository
see http://dev.mysql.com/doc/mysql/en/installing-source-tree.html

ChangeSet@stripped, 2006-12-14 16:41:24-05:00, iggy@alf. +1 -0
  Bug#18156 mysql_upgrade does not pass basedir or password arg to mysql and mysqlcheck
  - mysql_upgrade improperly reads the password and protocol options from command line.

  client/mysql_upgrade.c@stripped, 2006-12-14 16:40:41-05:00, iggy@alf. +5 -3
    Bug#18156 mysql_upgrade does not pass basedir or password arg to mysql and mysqlcheck
    - The password and protocol options were not being associate with their variables
    properly.

# This is a BitKeeper patch.  What follows are the unified diffs for the
# set of deltas contained in the patch.  The rest of the patch, the part
# that BitKeeper cares about, is below these diffs.
# User:	iggy
# Host:	alf.
# Root:	D:/src/mysql-5.0-maint_18156

--- 1.14/client/mysql_upgrade.c	2006-12-14 16:41:35 -05:00
+++ 1.15/client/mysql_upgrade.c	2006-12-14 16:41:35 -05:00
@@ -90,7 +90,8 @@
    (gptr*) &current_host, 0, GET_STR, REQUIRED_ARG, 0, 0, 0, 0, 0, 0},
   {"password", 'p',
    "Password to use when connecting to server. If password is not given"
-   " it's solicited on the tty.", 0, 0, 0, GET_STR, OPT_ARG, 0, 0, 0, 0, 0, 0},
+   " it's solicited on the tty.", (gptr*) &opt_password, (gptr*) &opt_password,
+   0, GET_STR, OPT_ARG, 0, 0, 0, 0, 0, 0},
 #ifdef __WIN__
   {"pipe", 'W', "Use named pipes to connect to server.", 0, 0, 0, 
    GET_NO_ARG, NO_ARG, 0, 0, 0, 0, 0, 0},
@@ -98,8 +99,9 @@
   {"port", 'P', "Port number to use for connection.", (gptr*) &opt_mysql_port,
    (gptr*) &opt_mysql_port, 0, GET_STR, REQUIRED_ARG, 0, 0, 0, 0, 0, 0},
   {"protocol", OPT_MYSQL_PROTOCOL,
-   "The protocol of connection (tcp,socket,pipe,memory).",
-   0, 0, 0, GET_STR, REQUIRED_ARG, 0, 0, 0, 0, 0, 0},
+   "The protocol of connection (tcp,socket,pipe,memory).", 
+   (gptr*) &opt_protocol, (gptr*) &opt_protocol, 0, GET_STR, REQUIRED_ARG, 0,
+   0, 0, 0, 0, 0},
 #ifdef HAVE_SMEM
   {"shared-memory-base-name", OPT_SHARED_MEMORY_BASE_NAME,
    "Base name of shared memory.", (gptr*) &shared_memory_base_name, 
